iac/environment/create_new_environment.py

```diff
@@ -42,13 +45,31 @@
     raise ValueError("The initial APIs must not be included in the services to enable")
 
 
+def _ensure_immutable_config(*, key: str, value: str):
+    past_outputs = get_pulumi_stack_outputs(stack_name=pulumi.get_stack(), module="environment")
+    locked_key = f"{key}_locked_at"
+    previous_value = past_outputs.get(locked_key)
+
+    if previous_value is not None and previous_value.value != value:
+        raise ValueError(f"Cannot change the value of {key} from {previous_value} to {value}")
+
+    pulumi.export(locked_key, value)
+
+
 def create_new_environment(*,
                            region: str,
                            realm_name: str,
                            folder_id: pulumi.Output[str],
                            billing_account: pulumi.Output[str],
+                           environment_type: EnvironmentTypes,
                            environment_name: str
                            ):
+    # Ensure that the environment name is immutable. This is to prevent, re-creating of the project.
```

Comment:
it is not the environment-name and realm-name that should be immutable, but the project_id.